routes.py

```diff
@@ -18,6 +18,10 @@
         handlers.api.UserHandler),
     (r"/api/(user)", handlers.api.UserHandler),
     (r"/api/shakes", handlers.api.UserShakesHandler),
+    (r"/api/shake_id/([0-9]+)",
+        handlers.api.ShakeHandler),
+    (r"/api/shake_name/([^a-zA-Z0-9\-]+)",
```

Comment:
I guess `shake_id` and `shake_name` should still be in parenthesis so they are sent as the value for `type`.